modules/network/pre-existing-subnetwork/main.tf

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+
+locals {
+  subnetwork_name = var.subnetwork_name != null ? var.subnetwork_name : var.subnetwork_self_link
```

Comment:
What is motivation behind this? 
It's fine not supply `name` as long as `self_link` is specified.
Let's remove it, unless there is a good reason for this.

Reply:
@mr0re1 I have changed the local name - it was an attempt to have something to display in the post condition, if the check fails.